pyccel/codegen/printing/ccode.py

```diff
@@ -430,14 +430,14 @@ def _print_LiteralComplex(self, expr):
             return self._print(PyccelAssociativeParenthesis(PyccelMul(expr.imag, LiteralImaginaryUnit())))
         else:
             return self._print(PyccelAssociativeParenthesis(PyccelAdd(expr.real,
-                            PyccelMul(expr.imag, LiteralImaginaryUnit()))))
+                            PyccelMul(expr.imag, LiteralImaginaryUnit()), simplify = True)))
 
     def _print_PythonComplex(self, expr):
         if expr.is_cast:
             value = self._print(expr.internal_var)
         else:
             value = self._print(PyccelAssociativeParenthesis(PyccelAdd(expr.real,
-                            PyccelMul(expr.imag, LiteralImaginaryUnit()))))
+                            PyccelMul(expr.imag, LiteralImaginaryUnit()), simplify = True)))
```

Comment:
This is similar to above. In order for anything to happen you also need to simplify `PyccelMul`. Here there is no risk of an infinite loop as it will go from `_print_PythonComplex` to `_print_LiteralComplex`, but we also don't gain anything by doing this